src/Reddit/Reddit.ts

```diff
@@ -105,208 +371,201 @@ export default <IPluginBase & IPlugin>{
     },
 
     init: async () => {
-      // there is a global command, so init runs everywhere
       if (document.location.hostname.endsWith("reddit.com")) {
-        console.log("init");
+        console.log("init")
 
-        if (/^https?:\/\/www.reddit/.test(document.location.href)) {
-          document.location.href = document.location.href.replace(
-            /^https?:\/\/.*\.reddit.com/,
-            "http://old.reddit.com"
-          );
-        }
+        isOldReddit = /https:\/\/old/.test(window.location.href)
 
-        if (COMMENTS_REGX.test(document.location.href)) {
-          PluginBase.util.prependContext("Post");
-          PluginBase.util.removeContext("Post List");
-        } else {
-          PluginBase.util.prependContext("Post List");
-          PluginBase.util.removeContext("Post");
-        }
+        toggleContext(COMMENTS_REGX.test(document.location.href))
 
-        await PluginBase.util.ready();
-
-        // number the elements
-        let index = 0;
-        for (let el of document.querySelectorAll<HTMLElement>(
-          "#siteTable>div.thing"
-        )) {
-          index++;
-          el.setAttribute(thingAttr, "" + index);
-          const rank = <HTMLElement>el.querySelector(".rank");
-          rank.setAttribute(
-            "style",
-            `
-                  display: block !important;
-                  margin-right: 10px;
-                  opacity: 1 !important';
-              `
-          );
-          rank.innerText = "" + index;
-        }
+        await PluginBase.util.ready()
+
+        window.addEventListener("load", onLoad)
+
+        setTimeout(() => {
+          if (!isDOMLoaded) {
+            const event = new Event("load")
+            window.dispatchEvent(event)
+          }
+        }, 2000)
       }
     },
 
     destroy: () => {
-      PluginBase.util.removeContext("Post List", "Post");
+      isDOMLoaded = false
+      PluginBase.util.removeContext("Post List", "Post")
+      observer && observer!.disconnect()
+      window.removeEventListener("load", onLoad)
     },
 
     commands: [
+      {
+        name: "Back",
```

Comment:
Can you explain what this is for? I'd rather not override the "Navigation/Go Back" command if it's avoidable, because otherwise it shows up under Reddit in the help and options.

Reply:
As I said, the "init" and "destroy" methods does not work for us when we go through the routes and we need to manually switch the commands context. If the user returns to the main screen via the "back" command, we will not be able to catch this moment and switch the context without redefinition.

Reply:
I think I understand now. But this means if the user uses the back button in the browser it will still break, no? Something like this would work better:
``` ts
window.addEventListener("locationchange", locationChangeHandler);

function locationChangeHandler() {
    // determine and switch to correct context here using window.location
}
```
The location change handler might not be fired even so. In that case, you'll need to add something like this as well:

``` ts
function attachScript(content: string) {
  const scriptEl = document.createElement("script");
  scriptEl.textContent = content;
  document.head.appendChild(scriptEl);
}
attachScript(
  `
        history.pushState = (f => function pushState(){
            let ret = f.apply(this, arguments);
            window.dispatchEvent(new Event('pushstate'));
            window.dispatchEvent(new Event('locationchange'));
            return ret;
        })(history.pushState);

        history.replaceState = (f => function replaceState(){
            let ret = f.apply(this, arguments);
            window.dispatchEvent(new Event('replacestate'));
            window.dispatchEvent(new Event('locationchange'));
            return ret;
        })(history.replaceState);
 `);
```

Reply:
if you haven't seen it already fixed it with listening to the "popstate" event